config/v1/types_infrastructure.go

```diff
@@ -614,6 +628,16 @@ type OpenStackPlatformStatus struct {
 	// datacenter DNS, a DNS service is hosted as a static pod to serve those hostnames
 	// to the nodes in the cluster.
 	NodeDNSIP string `json:"nodeDNSIP,omitempty"`
+
+	// loadBalancerType defines if a internal or external load balancer
+	// is to be used for the openshift api and ingress endpoints.
+	// When set to Internal the static pods in charge of API and Ingress traffic load-balancing
+	// defined in machine config operator will be deployed.
+	// When set to External these static pods will not be deployed.
+	// When not set, the default value is left to the cluster and currently defaults
+	// to Internal, so the static pods will be deployed.
+	// +kubebuilder:validation:Required
```

Comment:
Question for API reviewers:  do we want it to be required? Nothing else in `PlatformStatus` is required...

Reply:
> you should therefore make it optional else users will have to make the choice

ack, will make it optional.

> Given this is in the status, I would expect the only thing updating this to be the installer/a controller? Right?

Yes, correct.

Reply:
Oh. I thought that the correct way to set an enum optional with kubebuilder validations was to add the empty string as an option, and make the field required.

Reply:
@pierreprinetti I was told to follow this example: https://github.com/openshift/api/blob/master/example/v1/types_stable.go#L50-L54

Reply:
For spec, yes, that allows the value to be discovered, in status, since users don't need discovery, it's easier to omit the value